src/Microsoft.ML.Onnx/OnnxContextImpl.cs

```diff
@@ -246,6 +246,28 @@ public void AddInputVariable(ColumnType type, string colName)
             _inputs.Add(OnnxUtils.GetModelArgs(type, colName));
         }
 
+        /// <summary>
+        /// Retrieve the shape of an ONNX variable
+        /// </summary>
+        /// <param name="variableName">The ONNX name of the returned shape</param>
+        /// <returns>The shape of the retrieved variable</returns>
```

Comment:
This also seems to return `null` in some circumstances... is that expected?

Reply:
Yes, it's expected. Sometime, we define intermediate variables without shapes.